rfd/9999-oidc-idp-for-aws-rds-dbs-discovery.md

```diff
@@ -0,0 +1,495 @@
+5. Configures the role associated with that IdP
+   1. The user is asked to create or assign an existing role.
+   2. The role's `Policies` must be filled with the required Policies (described down below).
+   3. The role's `Trust relationships` must target the Identity Provider they just created (described down below).
+6. The user will be greeted with a message in Teleport saying that the configuration was successful.
+
+To ease the necessary manual steps, the Discover wizard will provide a small video demoing what the user must do for the previous steps.
+
+At this point, the integration is configured and the user is able to discover RDS DBs without leaving Teleport.
+
+#### System's Point of View
+The first interaction between AWS and Teleport happens when the user clicks on `Get Thumbprint` (step 4 of the User's Point of View flow).
+This will trigger a request started by AWS to Teleport hitting the OpenID Configuration endpoint and subsequently the JSON Web Key Set endpoint.
+
+After this step, when the user tries to list RDS DBs, Teleport generates a token and issues the API call.
+
+To generate a token, Teleport creates a JWT with the claims described above and the configured role, and signs it with the private key (the public is provided in the public JWKS endpoint).
+
+AWS receives the request and validates the token against the public key provided by OIDC IdP (ie, Teleport) at the `jwks_uri`.
+If authenticated and authorized for the api call, AWS returns the response to the API call.
+
+### Implementation Details
+
+#### Signing Key
+One of the requirements to be an OIDC provider is to provide the public key in a known HTTP endpoint and sign a JSON object (with claims) with the private key.
+
+We'll create a new RSA 2048 Key type and add it to the CertAuthority.
+This will be very similar to the JWTSigner key type used for App access.
+
+A single signing key will be used for this flow even if multiple AWS integrations are created (eg, multiple regions).
+
+#### HTTP Endpoints
+The following endpoints will be used during OIDC IdP set up:
+
+##### OpenID Configuration
+According to the [spec](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig) the Identity Provider must provide an endpoint at `<providerURL>/.well-known/openid-configuration` that returns the provider's configuration.
+
+So, a new endpoint at `<teleportProviderURL>/.well-known/openid-configuration` will be created that returns the following JSON:
+```
+200 OK
+
+{
+  "issuer": "<teleportProviderURL>", 
+  "jwks_uri": "<teleportProviderURL>/.well-known/jwks-oidc",
+  "claims": ["iss", "sub", "aud", "jti", "iat", "exp", "nbf"],
+  "id_token_signing_alg_values_supported": ["RS256"],
+  "response_types_supported": ["id_token"],
+  "scopes_supported": ["openid"],
+  "subject_types_supported": ["public", "pairwise"]
+}
+```
+
+- Issuer: the provider's URL
```

Comment:
This is basically proxy's public address, correct? Let's just call it that explicitly so it's clear.